modules/node/src/admin/admin.service.ts

```diff
@@ -177,4 +179,60 @@ export class AdminService {
 
     return toMerge;
   }
+
+  async fixProxyFactoryAddresses(): Promise<void> {
+    const channels = await this.channelService.findAll();
+
+    for (const channel of channels) {
+      const { data: stateChannel } = await this.cfCoreService.getStateChannel(
+        channel.multisigAddress,
+      );
+      console.log("stateChannel: ", stateChannel);
+      const contractAddresses = await this.configService.getContractAddresses();
+
+      const expectedMultisigAddress = await getCreate2MultisigAddress(
+        stateChannel.userNeuteredExtendedKeys,
+        stateChannel.proxyFactoryAddress || contractAddresses.ProxyFactory,
+        contractAddresses.MinimumViableMultisig,
+        this.configService.getEthProvider(),
+      );
+
+      if (expectedMultisigAddress === stateChannel.multisigAddress) {
+        if (!stateChannel.proxyFactoryAddress) {
+          logger.log(`Multisig address is as expected, adding correct proxyFactory address`);
+          // save new state channel
+        }
+      } else {
+        // try old proxy factory addresses to get expected address
+      }
+    }
+  }
+
+  async getIncorrectProxyFactoryAddresses(): Promise<string[]> {
+    const channels = await this.channelService.findAll();
+
+    const incorrectChannels = [];
+    for (const channel of channels) {
+      const { data: stateChannel } = await this.cfCoreService.getStateChannel(
+        channel.multisigAddress,
+      );
+      console.log("stateChannel: ", stateChannel);
+      const contractAddresses = await this.configService.getContractAddresses();
+      const expectedMultisigAddress = await getCreate2MultisigAddress(
+        stateChannel.userNeuteredExtendedKeys,
+        stateChannel.proxyFactoryAddress || contractAddresses.ProxyFactory,
+        contractAddresses.MinimumViableMultisig,
+        this.configService.getEthProvider(),
+      );
+
+      if (expectedMultisigAddress !== stateChannel.multisigAddress) {
+        logger.error(
+          `expectedMultisigAddress ${expectedMultisigAddress} not equal to stateChannel.multisigAddress ${stateChannel.multisigAddress}`,
+        );
+        incorrectChannels.push(stateChannel.multisigAddress);
```

Comment:
we should include the `bytecode` or factory address we found that *correctly* generates the multisig

Reply:
wrong place for this now that im looking at the whole file, my b